src/modules/users/gui/UsersPage.h

```diff
@@ -0,0 +1,143 @@
+/* === This file is part of Calamares - <http://github.com/calamares> ===
+ *
+ *   Copyright 2014-2015, Teo Mrnjavac <teo@kde.org>
+ *
+ *   Portions from the Manjaro Installation Framework
+ *   by Roland Singer <roland@manjaro.org>
+ *   Copyright (C) 2007 Free Software Foundation, Inc.
+ *
+ *   Calamares is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   Calamares is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with Calamares. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef USERSPAGE_H
+#define USERSPAGE_H
+
+#include "Typedefs.h"
+
+#include <QAbstractListModel>
+#include <QtGlobal>
+#include <QItemSelection>
+#include <QVariant>
+#include <QWidget>
+
+namespace Ui {
+class UserCreation;
+}
+
+/*
+ * Represents the properties of a user in the system.
+ *
+ * The fullname and avatar file can be empty and they will be ignored.
+ * The password can technically be empty too, but that is of course
+ * not recommended in general.
+ */
+struct User {
+    User(const QString& username, const QString& fullname, const QString& password, const QString& shell, const QString& avatarFile, bool autologin)
+        : username(username), fullname(fullname), password(password), shell(shell), avatarFile(avatarFile), autologin(autologin) {}
+
+    QString toString() const {
+        return username;
+    }
+
+    QString username;
+    QString fullname;
+    QString password;
+    QString shell;
+    QString avatarFile;
+    bool autologin;
+};
+
+/*
+ * Model to contain the users created during the installation
+ * process. This is associated to a QTableView in the main UI.
+ *
+ * Each row is a user and each column a property of the user.
+ */
+class UsersListModel : public QAbstractTableModel {
+    Q_OBJECT
+
+public:
+    virtual ~UsersListModel();
+
+    int rowCount(const QModelIndex &parent = QModelIndex()) const Q_DECL_OVERRIDE;
```

Comment:
No need to use `Q_DECL_OVERRIDE`, plain `override` is just fine since we depend on C++11 and Qt 5.